packages/devtools/test/integration_tests/server_test.dart

```diff
@@ -42,34 +59,180 @@ void main() {
   });
 
   test('registers service', () async {
-    // Track services as they're registered.
-    final registeredServices = <String>{};
+    final serverResponse =
+        await _send('vm.register', {'uri': appFixture.serviceUri.toString()});
+    expect(serverResponse['success'], isTrue);
 
-    // TODO(dantup): When the stable versions of Dart + Flutter are >= v3.22
-    // of the VM Service (July 2019), the _Service option here can be removed.
-    final serviceName = await appFixture.serviceConnection.serviceStreamName;
+    // Expect the VM service to see the launchDevTools service registered.
+    expect(registeredServices, contains('launchDevTools'));
+  }, timeout: const Timeout.factor(10));
 
-    appFixture.serviceConnection
-        .onEvent(serviceName)
-        .where((e) => e.kind == EventKind.kServiceRegistered)
-        .listen((e) => registeredServices.add(e.service));
+  test(
+      'DevTools connects back to server API and registers that it is connected',
+      () async {
+    // Register the VM.
+    await _send('vm.register', {'uri': appFixture.serviceUri.toString()});
+
+    // Send a request to launch DevTools in a browser.
+    await appFixture.serviceConnection
+        .callMethod(registeredServices['launchDevTools'], args: {});
+
+    final serverResponse = await _waitForClients(requiredConnectionState: true);
+    expect(serverResponse, isNotNull);
+    expect(serverResponse['clients'], hasLength(1));
+    expect(serverResponse['clients'][0]['hasConnection'], isTrue);
+    expect(serverResponse['clients'][0]['vmServiceUri'],
+        equals(appFixture.serviceUri.toString()));
+    // TODO(dantup): This test will fail until the devtools pubspec.yaml
```

Comment:
add a bug with this TODO or before landing break it up so you commit this integration test after committing the other changes on this cl.

Reply:
It's not just landing the change, in order to unskip this we need to push a new version of the server to pub, and then update the version number in devtools/pubspec.yaml. This is because the tests run having pulled devtools_server from pub, and not the version in the repo.

We could rig the tests to use the local version, but I think there's also value in testing against the pub version (after we've pushed this first version) to ensure we don't accidentally publish a DevTools that breaks against the live server (because it was only tested against unpublished code).

We could potentially run the tests twice, but I don't know how simple that'll be. Any preferences?

(If we land this as-is, I'll open an issue to unskip these after we next publish the server - it may be worthwhile doing dev builds of each just for some extra validation).